src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1345 @@
+            self::DEFAULT_CAPABILITIES[$browserName] ?? []
+        );
+        foreach ($defaults as $key => $value) {
+            if (is_null($caps->getCapability($key))) {
+                $caps->setCapability($key, $value);
+            }
+        }
+
+        // Merge in other requested types
+        foreach ($desiredCapabilities as $key => $value) {
+            $caps->setCapability($key, $value);
+        }
+
+        return $caps;
+    }
+
+    private function withSyn(): self
+    {
+        $hasSyn = $this->evaluateScript(
+            'return window.syn !== undefined && window.syn.trigger !== undefined'
+        );
+
+        if (!$hasSyn) {
+            $synJs = file_get_contents(__DIR__ . '/../resources/syn.js');
+            $this->webDriver->executeScript($synJs);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param int|string $char
+     * @throws JsonException
+     */
+    private function charToSynOptions($char, ?string $modifier = null): string
+    {
+        if (is_int($char)) {
+            $charCode = $char;
+            $char = chr($charCode);
+        } else {
+            $charCode = ord($char);
+        }
+
+        $options = [
+            'key' => $char,
+            'which' => $charCode,
+            'charCode' => $charCode,
+            'keyCode' => $charCode,
+        ];
+
+        if ($modifier) {
+            $options[$modifier . 'Key'] = true;
+        }
+
+        return json_encode($options, JSON_THROW_ON_ERROR);
+    }
+
+    /**
+     * Executes JS on a given element - pass in a js script string and argument[0] will
+     * be replaced with a reference to the result of the $xpath query
+     *
+     * @param string $xpath the xpath to search with
+     * @param string $script the script to execute
+     * @param bool $sync whether to run the script synchronously (default is TRUE)
+     *
+     * @return mixed
+     * @throws NoSuchElementException
+     * @example $this->executeJsOnXpath($xpath, 'return argument[0].childNodes.length');
+     */
+    private function executeJsOnXpath(
+        #[Language('XPath')]
+        string $xpath,
+        #[Language('JavaScript')]
+        string $script,
+        bool   $sync = true
+    ) {
+        return $this->executeJsOnElement($this->findElement($xpath), $script, $sync);
+    }
+
+    /**
+     * Executes JS on a given element - pass in a js script string and argument[0] will contain a reference to the element
+     *
+     * @param RemoteWebElement $element the webdriver element
+     * @param string $script the script to execute
+     * @param Boolean $sync whether to run the script synchronously (default is TRUE)
+     * @return mixed
+     * @example $this->executeJsOnXpath($xpath, 'return argument[0].childNodes.length');
+     */
+    private function executeJsOnElement(
+        RemoteWebElement $element,
+        #[Language('JavaScript')]
+        string           $script,
+        bool             $sync = true
```

Comment:
according to the code coverage, this method is never used in an async way. So it should be removed